### PR TITLE
JSON: encode NaNs and infinite floats as `null`

### PR DESCRIPTION
**Pull request by kamalmarhubi**
Such values are not allowed by the JSON spec. We match the behavior of
JSON.stringify in Firefox and Chrome.

fixes https://github.com/sandstorm-io/capnproto/issues/261

### CODE REVIEW

File: c++/src/capnp/compat/json.c++ [hunk 1]
```diff
@@ -237,9 +238,20 @@ void JsonCodec::encode(DynamicValue::Reader input, Type type, JsonValue::Builder
...
     case schema::Type::UINT16:
     case schema::Type::UINT32:
+      output.setNumber(input.as<double>());
+      break;
     case schema::Type::FLOAT32:
     case schema::Type::FLOAT64:
-      output.setNumber(input.as<double>());
+      {
+        double value = input.as<double>();
+        if (std::isinf(value) || std::isnan(value)) {
```
Comment (kentonv) [inline c++/src/capnp/compat/json.c++]: Please humor my NIH by using `kj::isNaN()`, and check for infinity by comparing for equality with `kj::inf()`. :)

File: c++/src/capnp/compat/json.h [hunk 2]
```diff
@@ -48,6 +48,8 @@ class JsonCodec {
   // - When encoding, all primitive fields are always encoded, even if default-valued. Pointer
   //   fields are only encoded if they are non-null.
   // - 64-bit integers are encoded as strings, since JSON "numbers" are double-precision floating
+  // - NaNs and infinite floating point numbers are not allowed by the JSON spec, and so are encoded
```
Comment (kentonv) [inline c++/src/capnp/compat/json.h]: You seem to have inserted this bullet point into the middle of another bullet point. :)

File: c++/src/capnp/compat/json.c++ (hunk 1 above)
Comment (kamalmarhubi) [inline c++/src/capnp/compat/json.c++]: Such NIH, wow. I had to introduce `kj::isNaN()` to do so. I'll send another tiny PR after this is merged to dedup its uses elsewhere.

File: c++/src/capnp/compat/json.h (hunk 2 above)
Comment (kamalmarhubi) [inline c++/src/capnp/compat/json.h]: Done.

**Pull request merged**